### PR TITLE
feat: add support for appengine legacy url encoding

### LINKED ISSUE #152: Query datastore entity by url-safe key or decode it to a searchable key
I'm trying to query datastore entity with NodeJS client API (https://www.npmjs.com/package/@google-cloud/datastore) given safe-url string provided by existing Go code. For some reason while other APIs (Go, Python,...) enable to decode/encode from/to safe-Url to key, NodeJS doesn't. Is there other way to do this? Thanks.

#### Environment details

  - OS:
  - Node.js version: 8.9.3
  - npm version: 6.2.0
  - `@google-cloud/datastore` version: 1.4.1

####

### PR DESCRIPTION
**Pull request by jiren**
Fixes #152 and #105 

- [x] Tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

**Comment by JustinBeckwith** [comment]
Thanks for the PR!  @crwilcox & @stephenplusplus could I trouble y'all to take a look here?

File: src/utils.ts [hunk 1]
```diff
@@ -0,0 +1,164 @@
+import * as Protobuf from 'protobufjs';
```
Comment (stephenplusplus) [inline src/utils.ts]: Could these new classes be moved to `src/entity.ts`?

Comment (jiren) [reply src/utils.ts]: ok

Comment (jiren) [reply src/utils.ts]: @stephenplusplus  Moved into entity.ts

**Comment by stephenplusplus** [comment]
@jiren looks like there is an issue with the samples test: https://source.cloud.google.com/results/invocations/f307d150-cef8-4d65-bb63-714d9931c311/targets/cloud-devrel%2Fclient-libraries%2Fnodejs%2Fpresubmit%2Fgoogleapis%2Fnodejs-datastore%2Fnode10%2Fsamples-test/log -- I'm not sure how this could be related to your changes, but other PRs don't seem to have the same failure. Any ideas?

**Comment by jiren** [comment]
@stephenplusplus  looking into it.

**Pull request merged**